### DISCUSSION TITLE
HeidiSQL works with `mysql-proxy` while DBeaver does not, and why?

### DISCUSSION THREAD

**Discussion by test-yzgoa** [dbeaver/dbeaver#22192] [Q&A]
For data analytical applications, our current development environment relies on a `MySQL-Proxy` (version 0.8) to connect to the MySQL data source.

Testing with all the latest versions, we noticed that:
- The proxy works OK with `HeidiSQL`. We noticed `HeidiSQL` uses driver `libmysql.dll` but we are not sure whether it's relevant to the system behavior difference.
- It does not work with `DBeaver`.

When not working, the symptom is that the client can complete password input, but then hangs forever upon opening the connection.

We wonder why `DBeaver` does not work, and if we can make it work. And we highly appreciate any hints and suggestions.

**Comment by LonwoLonwo** [comment]
Hello @test-yzgoa 

Did you try to add the libmysql.dll in the Connection Settings -> Driver Settings -> Libraries in DBeaver?

**Reply by test-yzgoa** [reply]
Added `libmysql.dll` by "Add File", see the screenshot below. However, the system behavior is still the same as before.
- The direct connection, e.g., to `192.168.56.yy` works in the same way without the DLL file, and
- The proxy connection, e.g., to `192.168.56.xx` still does not work.

Please let me know if I missed something here, and thank you.

![2023-12-18 11_56_17-Connection _proxy-test_ configuration](https://github.com/dbeaver/dbeaver/assets/153960794/a09b9dbc-1f46-434d-8483-5ce386d34f64)

**Reply by LonwoLonwo** [reply]
Did you try to change any parameters on the Proxy page?

![2023-12-19 11_06_02-Connect to a database](https://github.com/dbeaver/dbeaver/assets/45152336/f748c4cc-af6b-4f4c-be9e-747948799901)

**Reply by test-yzgoa** [reply]
I haven't tried the `Network configurations > Proxy` yet, however, I think this proxy is for network traffic only, so it's not the [`mysql-proxy`](https://downloads.mysql.com/docs/mysql-proxy-en.pdf) in our interest. If I missed anything, please let me know.

**Reply by LonwoLonwo** [reply]
Ok. Maybe the newest driver will help? https://mvnrepository.com/artifact/com.mysql/mysql-connector-j
You can add it to libraries. But do not forget to remove the old one (8.0.29) (you can return all back with the "reset to Defaults" button)
